### ISSUE TITLE
Pyright reports non-exhaustive match statement with Tuple[Any, Any]

### ISSUE THREAD

**Issue by kyle-query** [microsoft/pyright#3938]
**Describe the bug**

When reportMatchNotExhaustive is enabled (including typeCheckingMode = 'strict'), the type checker reports a match statement on `Tuple[Any, Any]` is not exhaustive, but with any other type parameters, eg `Tuple[int, int]`, the type checker correctly does not produce an error.

This only happens when the value is destructured, so matching on a variable (`case x: ...`) doesn't report an error.

**To Reproduce**

```python
# test.py
from typing import Any, Optional, Tuple

def f0(x: Tuple[Any, Any]) -> None:
    match x: # Cases within match statement do not exhaustively handle all values.
             #    Unhandled type: Tuple[Any, Any].
             #    If exhaustive handling is not intended, add "case _: pass"
        case (a,b): print(1)

def f1(x: Tuple[int, str]) -> None:
    match x: # no error
        case (a,b): print(1)
```

```toml
# pyproject.toml
[tool.pyright]
typeCheckingMode = "strict"
```

**Expected behavior**

The exhaustivity checker should not report an error.

**Screenshots or Code**

See above.

**VS Code extension or command-line**

```
» pyright --version
pyright 1.1.267
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.271, which I just published. It will also be included in a future release of pylance.

**Comment by kyle-query** [comment]
Thanks for the quick fix Eric!
